metaflow_extensions/utils.py

```diff
@@ -53,38 +56,43 @@ def running_in_local_env() -> bool:
     )
 
 
-def local_pip_install(project_path: Path) -> None:
-    """`pip install` local package at `project_path`."""
-    logging.info(f"pip installing @ {project_path} for {sys.executable}.")
+def is_mflow_conda_environment() -> bool:
+    """True if current process is a Metaflow Conda environment."""
+    return "--environment conda" in " ".join(sys.argv)
+
+
+def pip_install(executable: str, paths: Union[str, Tuple[str, ...]], *args) -> None:
+    """`pip install` packages corresponding to `paths`."""
     subprocess.run(
-        [
-            sys.executable,
-            "-m",
-            "pip",
-            "install",
-            str(project_path),
-            "--quiet",
-            # Needed to avoid temporary copies of large dirs like `outputs/`
-            "--use-feature",
-            "in-tree-build",
-        ],
+        _pip_install(executable, paths, *args),
         stdout=subprocess.DEVNULL,
         check=True,
     )
 
 
+def _pip_install(
+    executable: str, paths: Union[str, Tuple[str, ...]], *args
+) -> Tuple[str, ...]:
+    logging.info(f"pip installing {paths} for {executable}.")
+    if isinstance(paths, str):
+        paths = (paths,)
+
+    return (
+        executable,
+        "-m",
+        "pip",
+        "install",
+        *map(lambda x: shlex.quote(str(x)), paths),
+        "--quiet",
+        *map(lambda x: shlex.quote(str(x)), args),
+    )
```

Comment:
There will be some conflicts to negotiate here between this PR and #21

In order to install multiple libraries with `@pip(libraries: Dict[library, version] = {...})` then `path` needs to be `paths`.

I separated the subprocess call from the command building to be able to easily test the right command is built without having to test it's execution.

Reply:
yeah let's cross that bridge on a first come first served basis 😄, one person needs to refactor either way

Reply:
although if we know the order of PRs already then one person could rebase